### PR TITLE
Add CreateVMSnapshot to Provider interface

### PR DESCRIPTION
**Pull request by mansam**
This is laying some of the initial groundwork for Warm Migration.
https://github.com/kubevirt/vm-import-operator/issues/216

* Adds CreateVMSnapshot and SupportsWarmMigration to the Provider interface.
* Implements CreateVMSnapshot and SupportsWarmMigration for the vSphere provider.
* Stubs out CreateVMSnapshot and returns false for SupportsWarmMigration for the oVirt provider.

Signed-off-by: Sam Lucidi <slucidi@redhat.com>

### CODE REVIEW

**Comment by pkliczewski** [comment]
@mansam  please investigate CI failure:

> Error converting guests: virt-v2v job vmimport.v2v.kubevirt.io8dnlc failed}

**Comment by mansam** [comment]
This looks to be the same test failure from the machine type PR, but affecting a different test. I've run the test suite locally several times and have not yet been able to reproduce it.

/retest

**Comment by pkliczewski** [comment]
@mansam I do not think it is good idea to retest and move on. We need to understand what is causing the failure. How to you propose to track this issue? Should I open GH issue or BZ?

**Comment by pkliczewski** [comment]
BTW we need to branch before this one can be merged

**Comment by mansam** [comment]
I'm not trying to clear the failure and move on, I'm retesting in order to try to understand how the failure behaves. It's not consistently one test failing, and it doesn't always fail as we can see. Since I have not been able to reproduce this locally yet and there is not enough information in the Prow logs to identify the problem, I'm trying to get what information I can. I think https://github.com/kubevirt/vm-import-operator/issues/433 is probably appropriate for tracking this issue.

**Comment by ghost** [comment]
@pkliczewski, last comment is 2 weeks old and this PR is the first step to enable warm import. 
Besides the tests flakiness, do you have any objection to this PR?
You said that "we need to branch". Has the branching been done? If not, what is missing and can we help?

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
Discussed with @pkliczewski. This was waiting for a new release branch to be created. The `release-0.2` branch is now there, so merging this pull request.

**Pull request merged**
